File: src/crypto/keys/rsa/RsaPublicKey.ts
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import PublicKey from '../PublicKey';
import { KeyType } from '../KeyTypeFactory';

/**
 * Represents an RSA public key
 * @class
 * @extends PublicKey
 */
export default class RsaPublicKey extends PublicKey {
  /** 
   * Public exponent 
   */
  public e: string | undefined;
  /** 
   * Modulus 
   */
  public n: string | undefined;
  /**
   * Set the EC key type
   */
  kty = KeyType.RSA;
  /**
   * Set the default algorithm
   */
  alg = 'RS256';

  /**
   * Create instance of @class RsaPublicKey
   */
  constructor (key: any) {
    super(key)
    this.alg = key.alg;
    this.key_ops = key.key_ops;
    this.kid = key.kid;
    this.use = key.use;
    this.e = key.e;
    this.n = key.n;
  }
}

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/keys/rsa/RsaPublicKey.kt
package com.microsoft.did.sdk.crypto.keys.rsa

import com.microsoft.did.sdk.crypto.keys.KeyType
import com.microsoft.did.sdk.crypto.keys.PublicKey
import com.microsoft.did.sdk.crypto.models.webCryptoApi.JsonWebKey

class RsaPublicKey(jwk: JsonWebKey) : PublicKey(jwk) {
    override fun minimumAlphabeticJwk(): String {
        return "{\"e\":\"${e}\",\"kty\":\"${kty.value}\",\"n\":\"${n}\"}"
    }

    override var kty = KeyType.RSA

    val n = key.n
    val e = key.e

    override fun toJWK(): JsonWebKey {
        return JsonWebKey(
            kty = kty.value,
            alg = alg,
            use = use?.value,
            key_ops = key_ops?.map { use -> use.value },
            kid = kid,
            e = e,
            n = n
        )
    }
}